### PR TITLE
OSM Scout Nearby search

### PR DESCRIPTION
**Pull request by rinigus**
I have separated _poitype_ and _name_ in the _guide_ search of OSM Scout Server. This is to distinguish when the user wants to find a cafe by name or just any cafe. Updated API is at https://github.com/rinigus/osmscout-server#poi-search-near-a-reference-position . This API will be available with the next release of OSM Scout Server. 

The idea is to extend type support via rinigus/geocoder-nlp#37 and, for that, I would need to separate names and types in the query.

This PR adds _name_ as one of the trackable inputs and uses it to specify name component of the query to OSM Scout Server. It doesn't change anything for the search by others.

### CODE REVIEW

File: guides/osmscout.py [hunk 1]
```diff
@@ -44,6 +46,7 @@ def nearby(query, near, radius, params):
     """Return X, Y and a list of dictionaries of places matching `query`."""
     query = urllib.parse.quote_plus(query)
     limit = params.get("limit", 50)
+    name = params.get("name", "")
```
Comment (otsaloma) [inline guides/osmscout.py:49]: You probably need to add `name = urllib.parse.quote_plus(name)` here.

**Comment by otsaloma** [comment]
This is likely fine, I'll test it over the weekend.

Feels a bit odd though, is the reason you need two parameters that you expect false positives if you match names against a large list of type aliases? Taking both name and type in a single search field should work fine too, but maybe difficult to make discoverable for users?

**Comment by rinigus** [comment]
There are reasons for such split:

1. When I know which part is a type, I can normalize the string, trim it, and then just check against the current aliases (language specific). As a result, if I get the hits, I have the precise list of types (`amenity=cafe`). If I mix like 'Cafe Espresso House', that would be difficult to distinguish which part of the query is the name and which is a type. 

2. If I know the type, the search is a simple SQL query with the specified type. So, larger crunching goes through sqlite part. I still have to compare libpostal normalized names, if specified, but that should be user's specific request.

3. Already now, users could use by name or type. Not sure many of them know about it.

**Comment by otsaloma** [comment]
Tried it now. The only issue I see is that if someone would like to search by name only (apparently allowed by your API), it's not possible, but I can't think of a fix for that as in the normal case I want to prevent advancing with a blank type. Maybe with a good list of type aliases it won't be a problem? Otherwise I'm ready to merge.

**Comment by otsaloma** [comment]
Oh, and I have a release coming up soon – do I need to wait for a new release of your server or does the current version just gracefully ignore the excess name parameter?

**Comment by rinigus** [comment]
> Tried it now. The only issue I see is that if someone would like to search by name only (apparently allowed by your API), it's not possible, but I can't think of a fix for that as in the normal case I want to prevent advancing with a blank type. Maybe with a good list of type aliases it won't be a problem? 

I have been wondering about it too. I think, its not an issue and we can ask users to specify the type. List of aliases is at works (just finished importing them from https://wiki.openstreetmap.org/wiki/Nominatim/Special_Phrases) and will be available for users to see at https://github.com/rinigus/osmscout-server/blob/aliases/docs/tags/index.md (later will move to the master). So, let's stick to your design and just use the main part of the server API.

> Oh, and I have a release coming up soon – do I need to wait for a new release of your server or does the current version just gracefully ignore the excess name parameter?

It will ignore it. Server is pulling parameters from URL and anything that it doesn't know about is just ignored. So, this will be fine.

**Comment by otsaloma** [comment]
Thanks, merging.

**Pull request merged**